addons/pos_sale/static/src/js/models.js

```diff
@@ -33,6 +33,9 @@ const PosSaleOrderline = (Orderline) =>
             this.sale_order_line_id = this.sale_order_line_id || options.sale_order_line_id;
             this.down_payment_details = this.down_payment_details || options.down_payment_details;
             this.customerNote = this.customerNote || options.customer_note;
+            if (this.sale_order_origin_id && this.sale_order_origin_id.shipping_date) {
+                this.order.shipping_date = this.sale_order_origin_id.shipping_date;
```

Comment:
I'll prefer you calling the method you created to set the shipping date of the order instead of modifying its parents value. That way it's easier to maintain